src/com/esotericsoftware/kryo/util/Util.java

```diff
@@ -26,6 +26,9 @@
 /** A few utility methods, mostly for private use.
  * @author Nathan Sweet <misc@n4te.com> */
 public class Util {
+	//See: https://stackoverflow.com/questions/3038392/do-java-arrays-have-a-maximum-size
+	static public final int maxSafeArraySize = Integer.MAX_VALUE - 8;

```

Comment:
The issue here is that while you are certainly correct, the file being modified does not in fact utilize the standard (so it becomes a style issue: should the file be updated to match the standard or should it be left alone and made at least internally consistent?).

The line after the line the author added is:

`static public boolean isAndroid = "Dalvik".equals(System.getProperty("java.vm.name"));`

This should probably be changed to:

`public static final boolean IS_ANDROID = "Dalvik".equals(System.getProperty("java.vm.name"));`

... or something along those lines. Unless of course you intended for this to be modifiable... then you should probably have set/get and locks/synchronized access or make it an AtomicBoolean (or an AtomicBoolean abstracted by get/set so it can change later).

I do like the idea of minimal visibility and moving this to the IO package is probably the right call regardless of the naming standard and thread safety considerations I have raised.